### PR TITLE
[vcpkg_find_acquire_program] Update PKGCONFIG to 2.1.0

### PR DESCRIPTION
**Pull request by dg0yt**

### CODE REVIEW

**Comment by dg0yt** [comment]
Nicely complements #34228: pkgconf 2.1.0 seems to fix all issues of the 2.0 series.

File: scripts/cmake/vcpkg_find_acquire_program(PKGCONFIG).cmake [hunk 1]
```diff
@@ -21,16 +21,16 @@ elseif(CMAKE_HOST_WIN32)
...
             )
             set("${program}" "${PKGCONFIG_ROOT}/clangarm64/bin/pkg-config.exe" CACHE INTERNAL "")
         else()
             vcpkg_acquire_msys(PKGCONFIG_ROOT
                 NO_DEFAULT_PACKAGES
                 DIRECT_PACKAGES
-                    "https://mirror.msys2.org/mingw/mingw32/mingw-w64-i686-pkgconf-1~1.8.0-2-any.pkg.tar.zst"
-                    e5217d9c55ede4c15706b4873761cc6e987eabc1308120a3e8406571ae2993907f3776f2b2bba18d7aaec80ef97227696058cedc1b67a773530dc1e6077b95e6
+                    "https://mirror.msys2.org/mingw/mingw32/mingw-w64-i686-pkgconf-1~2.1.0-1-any.pkg.tar.zst"
+                    55626f0b7a6c950da75c4f7579b94859027f127c075042ab1f65b5387659eceb9e4c73dd3b79f54227772bd39a5759fbe10aa024cd38c8ac76677d0aec8458c1
```
Comment (MehdiChinoune) [inline scripts/cmake/vcpkg_find_acquire_program(PKGCONFIG).cmake:33]: Maybe use mingw32/mingw-w64-i686-pkgconf-1~2.1.0 on x86 and mingw64/mingw-w64-x86_64-pkgconf-1~2.1.0 on x64

**Comment by dg0yt** [comment]
Integrated in https://github.com/microsoft/vcpkg/pull/35331-

**Comment by dg0yt** [comment]
Merged with #35331.

**Pull request closed without merging**